### DISCUSSION TITLE
Accessing Config.toml document via Project API

### DISCUSSION THREAD

**Discussion by malinthar** [ballerina-platform/ballerina-lang#31789] [Q&A]
Currently [Project API](https://github.com/ballerina-platform/ballerina-lang/blob/997472a833aef268d7c21586543716f6f71ad179/compiler/ballerina-lang/src/main/java/io/ballerina/projects/Package.java) does not have a method returning the Config.toml document for a given package. To provide language support for Config.toml we may want to access the particular document. Is it possible to support this?

**Comment by IMS94** [comment]
IIRC, `Config.toml` is a runtime thing. Therefore, the Project API team may have a valid point to not provide an API to get that during the development time. Let's check what others have to say as well. @nadeeshaan @hasithaa @mohanvive

**Reply by nadeeshaan** [reply]
For the validation point of view, yes, this is a runtime validation and we will not be getting the diagnostics during the compilation time. Even though, if we consider the auto completions and similar language features, we have to access the tree model as we do for the `Ballerina.toml`.
@hevayo @azinneera would you be able shed some light on this requirement?
